nmostesting/mocks/Node.py

```diff
@@ -195,22 +196,37 @@ def parse_sdp(self, sdp_data):
             sdp_param_leg = {}
 
             media_line = re.search(r"m=([a-z]+) ([0-9]+) RTP/AVP ([0-9]+)", sdp_data)
-            sdp_param_leg["destination_port"] = int(media_line.group(2))
+            sdp_param_leg['destination_port'] = int(media_line.group(2))
 
             connection_line = re.search(r"c=IN IP[4,6] ([^/\r\n]*)(?:/[0-9]+){0,2}", sdp_data)
             destination_ip = connection_line.group(1)
             if ipaddress.IPv4Address(destination_ip).is_multicast:
-                sdp_param_leg["multicast_ip"] = destination_ip
-                sdp_param_leg["interface_ip"] = "auto"
+                sdp_param_leg['multicast_ip'] = destination_ip
+                sdp_param_leg['interface_ip'] = "auto"
             else:
-                sdp_param_leg["multicast_ip"] = None
-                sdp_param_leg["interface_ip"] = destination_ip
+                sdp_param_leg['multicast_ip'] = None
+                sdp_param_leg['interface_ip'] = destination_ip
 
             filter_line = re.search(r"a=source-filter: incl IN IP[4,6] (\S*) (\S*)", sdp_data)
             if filter_line and filter_line.group(2):
-                sdp_param_leg["source_ip"] = filter_line.group(2)
+                sdp_param_leg['source_ip'] = filter_line.group(2)
             else:
-                sdp_param_leg["source_ip"] = None
+                sdp_param_leg['source_ip'] = None
+
+            format_line = re.search(r"a=fmtp:(\S*\s*)(.*)", sdp_data)
+
+            if format_line and format_line.group(2):
+                #  Handle parameter keys that have no value, e.g. 'interlace' and set their value to 'True'
+                sdp_param_leg['format'] = {key_value.split('=')[0]:
+                                           key_value.split('=', maxsplit=1)[1] if '=' in key_value else True
+                                           for key_value in re.split(r'[ \t]*;[ \t]*', format_line.group(2))}
+
+                # Cast the string to an integer value for these parameters
+                int_params = {'packetmode', 'width', 'height', 'depth'}
```

Comment:
Hmm, `packetmode` is hard-coded in [_test_data/sdp/video-jxsv.sdp_](https://github.com/AMWA-TV/nmos-testing/pull/736/files#diff-782346a210916ef9e5f41c1cf5ff0c7a5e786245853fa7bdc0a004fdcb3c4ce5R11)

Reply:
Hmm, yeah - because the `packetmode` is in the format line it will be parsed and part of the returned dict.  So it's a question of whether to cast the value, ignore it, or even remove it from the dict.  Probably less confusing to just ignore it at this stage?

Reply:
OK, well, per the spec, it is a (0 or 1) integer then...

I have lost the plot on which `dict`s are used to populate an SDP template to make an SDP file, and which are created by parsing an SDP file or compared against that, or if any might be used to compare against IS-04 Source/Flow/Sender properties.

I leave it to you now! :-)